boa/src/builtins/json/mod.rs

```diff
@@ -38,19 +38,59 @@ mod tests;
 /// [spec]: https://tc39.es/ecma262/#sec-json.parse
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse
 // TODO: implement optional revever argument.
-pub fn parse(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+pub fn parse(_: &mut Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
     match serde_json::from_str::<JSONValue>(
         &args
             .get(0)
             .expect("cannot get argument for JSON.parse")
             .clone()
             .to_string(),
     ) {
-        Ok(json) => Ok(Value::from(json)),
+        Ok(json) => {
+            let j = Value::from(json);
+            match args.get(1) {
+                Some(reviver) if reviver.is_function() => {
+                    let mut holder = Value::new_object(None);
+                    holder.set_field(Value::from(""), j);
+                    walk(reviver, interpreter, &mut holder, Value::from(""))
+                }
+                _ => Ok(j),
+            }
+        }
         Err(err) => Err(Value::from(err.to_string())),
     }
 }
 
+/// This is a translation of the [Polyfill implementation][polyfill]
+/// This function recursively walks the structure. passing each key-value pair to the reviver function
+/// for possible transformation
+/// [polyfill]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse
```

Comment:
```suggestion
/// This is a translation of the [Polyfill implementation][polyfill]
///
/// This function recursively walks the structure. passing each key-value pair to the reviver function
/// for possible transformation
///
/// [polyfill]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse
```
Separate function title from  function description

Reply:
Done